packages/auth-types/index.d.ts

```diff
@@ -0,0 +1,281 @@
+  static PROVIDER_ID: string;
+  static credential(token: string): AuthCredential;
+}
+export class FacebookAuthProvider_Instance implements AuthProvider {
+  addScope(scope: string): AuthProvider;
+  providerId: string;
+  setCustomParameters(customOAuthParameters: Object): AuthProvider;
+}
+
+export class GithubAuthProvider extends GithubAuthProvider_Instance {
+  static PROVIDER_ID: string;
+  static credential(token: string): AuthCredential;
+}
+export class GithubAuthProvider_Instance implements AuthProvider {
+  addScope(scope: string): AuthProvider;
+  providerId: string;
+  setCustomParameters(customOAuthParameters: Object): AuthProvider;
+}
+
+export class GoogleAuthProvider extends GoogleAuthProvider_Instance {
+  static PROVIDER_ID: string;
+  static credential(
+    idToken?: string | null,
+    accessToken?: string | null
+  ): AuthCredential;
+}
+export class GoogleAuthProvider_Instance implements AuthProvider {
+  addScope(scope: string): AuthProvider;
+  providerId: string;
+  setCustomParameters(customOAuthParameters: Object): AuthProvider;
+}
+
+export class PhoneAuthProvider extends PhoneAuthProvider_Instance {
+  static PROVIDER_ID: string;
+  static credential(
+    verificationId: string,
+    verificationCode: string
+  ): AuthCredential;
+}
+export class PhoneAuthProvider_Instance implements AuthProvider {
+  constructor(auth?: FirebaseAuth | null);
+  providerId: string;
+  verifyPhoneNumber(
+    phoneNumber: string,
+    applicationVerifier: ApplicationVerifier
+  ): Promise<any>;
+}
+
+export class RecaptchaVerifier extends RecaptchaVerifier_Instance {}
+export class RecaptchaVerifier_Instance implements ApplicationVerifier {
+  constructor(
+    container: any | string,
+    parameters?: Object | null,
+    app?: FirebaseApp | null
+  );
+  clear(): any;
+  render(): Promise<any>;
+  type: string;
+  verify(): Promise<any>;
+}
+
+export class TwitterAuthProvider extends TwitterAuthProvider_Instance {
+  static PROVIDER_ID: string;
+  static credential(token: string, secret: string): AuthCredential;
+}
+export class TwitterAuthProvider_Instance implements AuthProvider {
+  providerId: string;
+  setCustomParameters(customOAuthParameters: Object): AuthProvider;
+}
+
+export type UserCredential = {
+  additionalUserInfo?: AdditionalUserInfo | null;
+  credential: AuthCredential | null;
+  operationType?: string | null;
+  user: User | null;
+};
+
+export interface UserMetadata {
+  creationTime?: string;
+  lastSignInTime?: string;
+}
+
+export type Persistence = string;
+
+export class FirebaseAuth {
+  private constructor();
+
+  static Persistence: {
+    LOCAL: Persistence;
+    NONE: Persistence;
+    SESSION: Persistence;
+  };
+
+  app: FirebaseApp;
+  applyActionCode(code: string): Promise<any>;
+  checkActionCode(code: string): Promise<any>;
+  confirmPasswordReset(code: string, newPassword: string): Promise<any>;
+  createUserWithEmailAndPassword(email: string, password: string): Promise<any>;
```

Comment:
Can you add the following 4 methods that are being released this week:
`createUserAndRetrieveDataWithEmailAndPassword(email: string, password: string): Promise<any>;`

Reply:
Done